packages/hop-node/src/1inch/1inch.ts

```diff
@@ -0,0 +1,347 @@
+    }
+
+    return json
+  }
+
+  async getQuote (params: QuoteParams) {
+    const { fromTokenAddress, toTokenAddress, amount } = params
+    if (!this.chainId) {
+      throw new Error('chainId is required')
+    }
+    if (!fromTokenAddress) {
+      throw new Error('fromTokenAddrss is required')
+    }
+    if (!toTokenAddress) {
+      throw new Error('toTokenAddress is required')
+    }
+    if (!amount) {
+      throw new Error('amount is required')
+    }
+
+    const url = this.constructUrl('/quote', {
+      fromTokenAddress,
+      toTokenAddress,
+      amount
+    })
+    const result = await this.getJson(url)
+    if (!result.toTokenAmount) {
+      logger.error(result)
+      throw new Error('expected tx data')
+    }
+
+    const { toTokenAmount } = result
+
+    return toTokenAmount
+  }
+
+  async getAllowance (params: AllowanceParams) {
+    const { tokenAddress, walletAddress } = params
+    if (!this.chainId) {
+      throw new Error('chainId is required')
+    }
+    if (!tokenAddress) {
+      throw new Error('tokenAddress required')
+    }
+    if (!walletAddress) {
+      throw new Error('walletAddress is required')
+    }
+
+    const url = this.constructUrl('/approve/allowance', {
+      tokenAddress,
+      walletAddress
+    })
+    const result = await this.getJson(url)
+
+    if (result.allowance === undefined) {
+      logger.error(result)
+      throw new Error('expected tx data')
+    }
+
+    return result.allowance
+  }
+
+  async getApproveTx (params: ApproveParams) {
+    const { tokenAddress, amount } = params
+    if (!this.chainId) {
+      throw new Error('chainId is required')
+    }
+    if (!tokenAddress) {
+      throw new Error('tokenAddress required')
+    }
+    if (!amount) {
+      throw new Error('amount is required')
+    }
+
+    const url = this.constructUrl('/approve/transaction', {
+      amount,
+      tokenAddress
+    })
+    const result = await this.getJson(url)
+
+    if (!result.data) {
+      logger.error(result)
+      throw new Error('expected tx data')
+    }
+
+    const { data, to, value } = result
+
+    return {
+      data,
+      to,
+      value
+    }
+  }
+
+  async getSwapTx (params: SwapParams) {
+    const { fromTokenAddress, toTokenAddress, fromAddress, amount, slippage, destReceiver } = params
+    if (!this.chainId) {
+      throw new Error('chainId is required')
+    }
+    if (!fromTokenAddress) {
+      throw new Error('fromTokenAddrss is required')
+    }
+    if (!toTokenAddress) {
+      throw new Error('toTokenAddress is required')
+    }
+    if (!fromAddress) {
+      throw new Error('fromAddress is required')
+    }
+    if (!amount) {
+      throw new Error('amount is required')
+    }
+    if (!slippage) {
+      throw new Error('slippage is required')
+    }
+
+    const url = this.constructUrl('/swap', {
+      fromTokenAddress,
+      toTokenAddress,
+      amount,
+      fromAddress,
+      slippage,
```

Comment:
I see we use `0.1` slippage by default. Just confirming, do you know if this endpoint expect the slippage in whole numbers or percentage (i.e. does it expect our `0.1` value or is it looking for `0.001`)?